Bottom: 585026a4baecf1fdc9c3be4119abadd61d7e943b
Top:    b9c19fb4396d0df37bda5f1c4ecb6d52890c3112
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-09-13 15:41:47 -0500

haproxy configuration



---
diff --git a/ansible-playbook/group_vars/docker-hosts/networking.yml b/ansible-playbook/group_vars/docker-hosts/networking.yml
index f652891a..b73f26e2 100644
--- a/ansible-playbook/group_vars/docker-hosts/networking.yml
+++ b/ansible-playbook/group_vars/docker-hosts/networking.yml
@@ -1,30 +1,20 @@
 ---
 
 svn_hostname: svn.novatech-llc.com
-svn_port_args:
-- "{{ svn_ip_addr }}:80:80"
-- "{{ svn_ip_addr }}:443:443"
+svn_port_args: []
 
 gitlab_hostname: git.novatech-llc.com
 gitlab_port_args:
-- "{{ gitlab_ip_addr }}:80:80"
-- "{{ gitlab_ip_addr }}:443:443"
 - "{{ gitlab_ip_addr }}:22:22"
 
 wiki_hostname: wiki.novatech-llc.com
-wiki_port_args:
-- "{{ wiki_ip_addr }}:80:80"
-- "{{ wiki_ip_addr }}:443:443"
+wiki_port_args: []
 
 mantisbt_hostname: mantis.novatech-llc.com
-mantisbt_port_args:
-- "{{ mantisbt_ip_addr }}:80:80"
-- "{{ mantisbt_ip_addr }}:443:443"
+mantisbt_port_args: []
 
 build_hostname: buildsystem.novatech-llc.com
-build_port_args:
-- "{{ build_ip_addr }}:80:80"
-- "{{ build_ip_addr }}:443:443"
+build_port_args: []
 
 testssh_hostname: testssh.novatech-llc.com
 testssh_port_args:
@@ -32,9 +22,7 @@ testssh_port_args:
 
 exim4_hostname: mail.novatech-llc.com
 exim4_port_args:
-- "{{ exim4_ip_addr }}:25:25"
-- "{{ exim4_ip_addr }}:465:465"
-- "{{ exim4_ip_addr }}:587:587"
+- "25"
 
 openldap_hostname: ldap.novatech-llc.com
 openldap_port_args:
@@ -42,19 +30,13 @@ openldap_port_args:
 - "{{ openldap_ip_addr }}:636:636"
 
 phpldapadmin_hostname: ldap.novatech-llc.com
-phpldapadmin_port_args:
-- "{{ phpldapadmin_ip_addr }}:80:80"
-- "{{ phpldapadmin_ip_addr }}:443:443"
+phpldapadmin_port_args: []
 
 phpmyadmin_hostname: phpmyadmin.novatech-llc.com
 
 buildbot_hostname: buildbot.novatech-llc.com
 buildbot_port_args:
-- "{{ buildbot_ip_addr }}:80:8080"
-- "{{ buildbot_ip_addr }}:443:8443"
 - "{{ buildbot_ip_addr }}:9989:9989"
 
 nexus_hostname: nexus.novatech-llc.com
-nexus_port_args:
-- "{{ nexus_ip_addr }}:80:8081"
-- "{{ nexus_ip_addr }}:443:8443"
+nexus_port_args: []
diff --git a/roles/haproxy/files/haproxy.cfg b/roles/haproxy/files/haproxy.cfg
new file mode 100644
index 00000000..eccf2ffc
--- /dev/null
+++ b/roles/haproxy/files/haproxy.cfg
@@ -0,0 +1,93 @@
+global
+    daemon
+    maxconn 256
+    ssl-dh-param-file /etc/ssl/private/dhparam.pem
+
+defaults
+    mode http
+    timeout connect 5s
+    timeout client 50s
+    timeout server 50s
+    default-server init-addr last,libc,none
+
+#enable stats page
+listen stats
+    bind *:9000
+    mode http
+    stats enable
+
+#
+frontend container_https_in
+    mode http
+    bind *:80
+    bind *:443 ssl crt /etc/ssl/private/haproxy.pem
+
+    http-request set-header X-Forwarded-Proto https
+    http-request set-header X-SSL %[ssl_fc]
+
+    http-request redirect scheme https code 301 if !{ ssl_fc }
+
+    acl is_buildbot     hdr(host) eq buildbot.novatech-llc.com
+    acl is_gitlab       hdr(host) eq git.novatech-llc.com
+    acl is_ldapadmin    hdr(host) eq ldap.novatech-llc.com
+    acl is_mantisbt     hdr(host) eq mantis.novatech-llc.com
+    acl is_nexus        hdr(host) eq nexus.novatech-llc.com
+    acl is_svn          hdr(host) eq svn.novatech-llc.com
+    acl is_wiki         hdr(host) eq wiki.novatech-llc.com
+
+    use_backend backend-buildbot if is_buildbot
+    use_backend backend-nexus if is_nexus
+    use_backend backend-svn if is_svn
+    use_backend backend-ldapadmin if is_ldapadmin
+    use_backend backend-gitlab if is_gitlab
+    use_backend backend-mantisbt if is_mantisbt
+    use_backend backend-wiki if is_wiki
+
+# frontend container_ldap_in
+#     mode tcp
+#     bind *:389
+#     bind *:636
+#     use_backend backend-ldap
+
+# frontend container_smtp_in
+#     mode tcp
+#     bind *:25
+#     bind *:465
+#     bind *:587
+#     use_backend backend-exim4
+
+# frontend container_ssh_in
+#     mode tcp
+#     bind *:22
+#     default_backend
+
+#
+backend backend-svn
+    server svn1 svn.frontend:80
+
+backend backend-ldapadmin
+    server phpldapadmin1 phpldapadmin.frontend:80
+
+backend backend-exim4
+    server exim4_1 exim4.frontend:25
+
+backend backend-gitlab
+    server gitlab1 gitlab.frontend:80
+
+backend backend-wiki
+    server wiki1 wiki.frontend:80
+
+backend backend-mantisbt
+    server mantisbt1 mantisbt.frontend:80
+
+backend backend-build
+    server build1 build.frontend:80
+
+backend backend-testssh
+    server testssh1 testssh.frontend:80
+
+backend backend-buildbot
+    server buildbot1 buildbot.frontend:80
+
+backend backend-nexus
+    server nexus1 nexus3.frontend:8081
